### PR TITLE
Implemented Search Fragment and Added Explore Page

### PR DESCRIPTION
**Pull request by ryankortbeek**
- implemented SearchDialogFragment
- implemented fragment_search_dialog.xml
- implemented ExploreActivity
- implemented activity_explore.xml
- clicking the search button from any page now will bring up the search fragment
    - upon the user clicking cancel the fragment will simply close
    - upon the user clicking confirm the fragment will verify that the search bar is not empty (if it is empty it will prompt the user to enter a keyword and will remain open), and then will take the user to the Explore page where the search results will be displayed
- all pages that have a search icon in the menu bar on top now support the functionality mentioned above (this can be changed on a per-page basis as each one of these pages (ExploreActivity, MyBooksActivity, BorrowedActivity, and PendingRequestsActivity) implements the executeSearch from the OnFragmentInteractionListener interface that is part of the SearchDialogFragment class) 

Relevant issue(s):
- #11 
- #17

### CODE REVIEW

**Comment by ryankortbeek** [comment]
Note all the functionality up to this point (including the last 2 pull requests) has been confirmed to work as expected (no formal tests have been written yet though)

File: src/app/src/main/java/com/example/bookmark/fragments/SearchDialogFragment.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        // Available books being all books that are not currently accepted or
+        // borrowed
+        builder.setView(view).setTitle("Search Available Books").setNegativeButton(
+            "CANCEL", null).setPositiveButton("CONFIRM", null);
+        final AlertDialog dialog = builder.create();
+        dialog.show();
+        dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                boolean proceed = Boolean.TRUE;
```
Comment (kylehennig) [inline src/app/src/main/java/com/example/bookmark/fragments/SearchDialogFragment.java:66]:
```suggestion
                boolean proceed = true;
```

File: src/app/src/main/java/com/example/bookmark/fragments/SearchDialogFragment.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                // There must be some text in the search bar to execute a search
+                if (TextUtils.isEmpty(searchEditText.getText())) {
+                    searchEditText.setError("Please enter the keywords that " +
+                        "you would like to search.");
+                    proceed = Boolean.FALSE;
```
Comment (kylehennig) [inline src/app/src/main/java/com/example/bookmark/fragments/SearchDialogFragment.java:71]:
```suggestion
                    proceed = false;
```

**Review by kylehennig — CHANGES_REQUESTED** [review]
`Boolean.TRUE` and `Boolean.FALSE` should be used for `Boolean` types. Primitive `boolean` types (note the lowercase b) should be set equal to `true` or `false`.

File: src/app/src/main/java/com/example/bookmark/fragments/SearchDialogFragment.java (hunk 2 above)
Comment (ryankortbeek) [reply src/app/src/main/java/com/example/bookmark/fragments/SearchDialogFragment.java:71]: Good catch lol I don't know why I had this

**Review by kylehennig — APPROVED** [review]

**Pull request merged**